initialize/initialize.go

```diff
@@ -155,12 +163,18 @@ func githubSearch(query string) ([]Repo, error) {
 		opts.Page = resp.NextPage
 	}
 
+	hostname := "github.com"
+	if os.Getenv("GITHUB_URL") != "" {
+		baseEndpoint, _ := url.Parse(os.Getenv("GITHUB_URL"))
+		hostname = baseEndpoint.Hostname()
+	}
+
 	repos := []Repo{}
 	for _, r := range allRepos {
 		repos = append(repos, Repo{
 			Name:     r.GetName(),
 			Owner:    r.Owner.GetLogin(),
-			CloneURL: fmt.Sprintf("git@github.com:%s", r.GetFullName()),
+			CloneURL: fmt.Sprintf("git@%s:%s", hostname, r.GetFullName()),
 			Provider: "github",
```

Comment:
Ultimately, I think I would like to refactor this a bit.. No need to do that within this PR but what do you think of the following? 

I believe a given microplane flow is with only one provider. If so, then we can abstract out the idea of a provider a bit more, make it more explicit, and fix edge cases (e.g. currently, it's possible that the env var value could change and the API URL would be different between `push` and `merge`!)

In the future, a user might run: `mp init --provider github --provider-url <your API URL>` (these flags would be optional and have defaults) 

Then, we can store the provider and providerURL on `initialize.Output`
https://github.com/Clever/microplane/blob/f9bb2bd50cb558749c4023c518bea701747cf561/initialize/initialize.go#L34-L39

When doing all future steps, we use the stored provider values. e.g. we can make `CloneURL()` a function  that generates a string based on provider values, and we can remove URL parsing logic in later steps.

Reply:
This is a great idea! I didn't think about what would happen if someone changed `GITHUB_URL` part way through the process. It should definitely be made idempotent.